src/Components/ProductList/ProductList.js

```diff
@@ -0,0 +1,50 @@
+import { useEffect } from "react";
+import axios from "axios";
+import { ProductCard } from "../ProductCard/ProductCard";
+import { useProducts } from "../../contexts/product-context";
+import { getFilteredProducts } from "../../utilities";
+
+export const ProductList = () => {
+  const {
+    state: { products, sortBy, rating, categories, priceRange },
+    dispatch,
+  } = useProducts();
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const productResponse = await axios.get("/api/products");
+        dispatch({
+          type: "LOAD_PRODUCTS",
+          payload: productResponse.data.products,
+        });
+      } catch (error) {
+        console.error(error);
+      }
+    })();
+  }, [dispatch]);
+
+  useEffect(() => {
+    document.title = "Products | Echo Store";
```

Comment:
This was an amazing way to update the title of the webpage.
I shall implement this in my app as well.
Thanks